tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestExampleGenerator.java

```diff
@@ -1123,6 +1123,56 @@ private void checkDayPeriod(
         checkPathValue(exampleGenerator, path, cldrFile.getStringValue(path), expected);
     }
 
+    public void TestAllDayPeriods() { 
+        // excludes midnight, see ICU-12278 
+        String[][] tests = {
```

Comment:
Yay! I love this test. My advice is only on style -- I think this test gets the job done well. I know many other tests in this file follow this paradigm but I'm personally allergic to it :p so feel free to ignore my advice or change it.

String[][] types are very unsafe -- it's not clear what the strings in the middle represent and they have undetermined length.

Rather, a better idea would be to make a new method like to `checkValue` -- but specific to the kind of test you are running `checkDayPeriodsForLocale`
https://github.com/unicode-org/cldr/blob/cf294f071f911fc751b1af5fff7a0410a7ab2a70/tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestExampleGenerator.java#L454-L461

In that test you will include the code that you have now in your for loop down below.

And then you will use it individually for each example, and now things will have nice variable names
https://github.com/unicode-org/cldr/blob/cf294f071f911fc751b1af5fff7a0410a7ab2a70/tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestExampleGenerator.java#L447-L452

Reply:
Thanks for the pointer! I didn't know that about String[][], but your explanation makes sense and I'll revise the tests adding a `checkValue` type method instead.